### ISSUE TITLE
Equihash Continuous Restart when switching Algos.

### ISSUE THREAD

**Issue by Justaphf** [nicehash/NiceHashMiner-Archived#361]
For some reason it seems that when profitability changes such that a change of mining algorithm is prudent, if the now most profitable algorithm is Equihash, the client tries to start many instances of the equihash miner.  All of these instances seem to instantly fail and new instances are attempted.  This is portrayed as a continual series of CMD windows popping up and disappearing in less than a second.

This is CPU related, but only when switching between algorithms where the "to" algo is Equihash.  Upon starting the node, if the start algo is Equihash everything is fine, it's just when switching from cryptonight back to Equihash that the issue shows up.

For reference:
GPU;  EVGA GTX 1070 SC Black
CPU: i5-3570K (~3-5 years old, so older batch).

EDIT: Clarity & typos.

**Comment by Justaphf** [comment]
Upon further investigation is is definiately the CPU, the GPU stays on daggerhasimoto the whole time.  The issue seems to be occuring in the Miner.cs file, function `Reset()`.  The starts seem to keep bouncing between ports 4002 and 4004.

Subsection of the output logs below for reference:

```
1:42:41] [MiningSession] Current Global profit: 1.72319958 USD/Day
[01:42:41] [MiningSession] Current Global profit: IS PROFITABLE MinProfit mine always regardless of profit
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Changing miner port from 4004 to 4002
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Starting miner [pid(11628)|bin(bin\eqm\eqm.exe)] --algo=equihash --url=equihash.usa.nicehash.com:3357 --userpass=_btcaddr_.debug1:x --threads=4 --api-bind=4002
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(11628)|bin(bin\eqm\eqm.exe)] Starting cooldown checker
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(11628)|bin(bin\eqm\eqm.exe)] Manually closed, will restart
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(11628)|bin(bin\eqm\eqm.exe)] Restarting miner..
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(11628)|bin(bin\eqm\eqm.exe)] Shutting down miner
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Changing miner port from 4002 to 4004
The thread 0x4f0 has exited with code 0 (0x0).
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Starting miner [pid(9060)|bin(bin\eqm\eqm.exe)] --algo=equihash --url=equihash.usa.nicehash.com:3357 --userpass=_btcaddr_.debug1:x --threads=4 --api-bind=4004
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9060)|bin(bin\eqm\eqm.exe)] Starting cooldown checker
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9060)|bin(bin\eqm\eqm.exe)] Manually closed, will restart
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9060)|bin(bin\eqm\eqm.exe)] Restarting miner..
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9060)|bin(bin\eqm\eqm.exe)] Shutting down miner
[01:42:41] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Changing miner port from 4004 to 4002
[01:42:42] [ExtraLaunchParametersParser] ExtraLaunch params "-w worker1 --threads=4" for device UUID GEN-e7b9e377fe88b42d13cfb690252bfb3fc1c19a92c136aa9ef0c71721c1ddbcae
[01:42:42] [ExtraLaunchParametersParser] Cannot parse "-w", not supported, set to ignore, or wrong extra launch parameter settings
[01:42:42] [ExtraLaunchParametersParser] Cannot parse "worker1", not supported, set to ignore, or wrong extra launch parameter settings
[01:42:42] [ExtraLaunchParametersParser] Final extra launch params parse " --threads=4"
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Starting miner [pid(9776)|bin(bin\eqm\eqm.exe)] --algo=equihash --url=equihash.usa.nicehash.com:3357 --userpass=_btcaddr_.debug1:x --threads=4 --api-bind=4002
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9776)|bin(bin\eqm\eqm.exe)] Starting cooldown checker
[01:42:42] [NICEHASH] SMA get
The thread 0x2524 has exited with code 0 (0x0).
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9776)|bin(bin\eqm\eqm.exe)] Manually closed, will restart
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9776)|bin(bin\eqm\eqm.exe)] Restarting miner..
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(9776)|bin(bin\eqm\eqm.exe)] Shutting down miner
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Changing miner port from 4002 to 4004
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Starting miner [pid(6308)|bin(bin\eqm\eqm.exe)] --algo=equihash --url=equihash.usa.nicehash.com:3357 --userpass=_btcaddr_.debug1:x --threads=4 --api-bind=4004
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(6308)|bin(bin\eqm\eqm.exe)] Starting cooldown checker
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(6308)|bin(bin\eqm\eqm.exe)] Manually closed, will restart
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(6308)|bin(bin\eqm\eqm.exe)] Restarting miner..
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(6308)|bin(bin\eqm\eqm.exe)] Shutting down miner
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Changing miner port from 4004 to 4002
The thread 0xd18 has exited with code 0 (0x0).
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Starting miner [pid(10064)|bin(bin\eqm\eqm.exe)] --algo=equihash --url=equihash.usa.nicehash.com:3357 --userpass=_btcaddr_.debug1:x --threads=4 --api-bind=4002
[01:42:42] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(10064)|bin(bin\eqm\eqm.exe)] Starting cooldown checker
The thread 0x988 has exited with code 0 (0x0).
[01:42:43] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(10064)|bin(bin\eqm\eqm.exe)] Manually closed, will restart
[01:42:43] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(10064)|bin(bin\eqm\eqm.exe)] Restarting miner..
[01:42:43] [CPU-MINER_ID(2)-DEVICE_IDs(0)] [pid(10064)|bin(bin\eqm\eqm.exe)] Shutting down miner
[01:42:43] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Changing miner port from 4002 to 4004
[01:42:43] [CPU-MINER_ID(2)-DEVICE_IDs(0)] Starting miner [pid(9388)|bin(bin\eqm\eqm.exe)] --algo=equihash --url=equihash.usa.nicehash.com:3357 --userpass=_btcaddr_.debug1:x --threads=4 --api-bind=4004
```

**Comment by Justaphf** [comment]
Additional investigation seems to indicate that the root cause is related to the user preference as to whether or not mining windows should be hidden or not.  If mining windows are marked as should be hidden, then the issue occurs when a switch from one mining algo to equihash occurs.

However, if algo mining windows are marked to be displayed via the preferences, then the switch occurs smoothly.

**Comment by ferdjones** [comment]
I have seen this issue also. The only way I can get it to stop is to go into task manager and kill the Nicehashminer process.  If I restart nicehashminer it will start mining equihash without a problem, BUT, it ignores the hide mining windows preference.

Also, I get the following when the equihash miner starts:
(i7-6900k, GTX 1080, equihash set to 4 less threads)

```
==================== www.nicehash.com ====================
     EQM v1.0.1b equihash CPU&GPU Miner for NiceHash.
  This miner is using proprietary base code by NiceHash.
    CPU solver: by xenoncat
    CUDA solver: by djeZo; heavily modified Tromp solver
==================== www.nicehash.com ====================

[07:38:34][0x00004604][info] Log started
[07:38:34][0x00004604][info] Miner starting up, press [CTRL + C] to quit
[07:38:34][0x00004604][info] net | Connecting to private.equihash.usa.nicehash.com:3330
[07:38:34][0x00000c14][info] miner#0 | Initializing thread #0 (CPU-XENONCAT-AVX2)
[07:38:34][0x00001504][info] miner#1 | Initializing thread #1 (CPU-XENONCAT-AVX2)
[07:38:34][0x000018ac][info] miner#2 | Initializing thread #2 (CPU-XENONCAT-AVX2)
[07:38:34][0x00004054][info] miner#3 | Initializing thread #3 (CPU-XENONCAT-AVX2)
[07:38:34][0x00003a6c][info] miner#4 | Initializing thread #4 (CPU-XENONCAT-AVX2)
[07:38:34][0x00000944][info] miner#5 | Initializing thread #5 (CPU-XENONCAT-AVX2)
[07:38:34][0x00003838][info] miner#6 | Initializing thread #6 (CPU-XENONCAT-AVX2)
[07:38:34][0x000011f4][info] miner#7 | Initializing thread #7 (CPU-XENONCAT-AVX2)
[07:38:34][0x000020d0][info] miner#8 | Initializing thread #8 (CPU-XENONCAT-AVX2)
[07:38:34][0x00003420][info] miner#9 | Initializing thread #9 (CPU-XENONCAT-AVX2)
[07:38:34][0x0000470c][info] miner#10 | Initializing thread #10 (CPU-XENONCAT-AVX2)
[07:38:34][0x0000401c][info] miner#11 | Initializing thread #11 (CPU-XENONCAT-AVX2)
[07:38:34][0x00004604][info] api | Listening on port 4001
[07:38:34][0x00000c14][warning] miner#0 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x00001504][warning] miner#1 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x000018ac][warning] miner#2 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x00004054][warning] miner#3 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x00003a6c][warning] miner#4 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x00000944][warning] miner#5 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x00003838][warning] miner#6 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x000011f4][warning] miner#7 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x000020d0][warning] miner#8 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x00003420][warning] miner#9 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x0000470c][warning] miner#10 | Unable to alloc locked pages (err=1314); performance will be degraded
[07:38:34][0x0000401c][warning] miner#11 | Unable to alloc locked pages (err=1314); performance will be degraded
```

**Comment by Justaphf** [comment]
This seems fixed as of version 1.7.3.6.  The equihash miner doesn't respect the "don't show miner windows" setting, but all other miners do, and the constant restart of many instances is gone.

Marking as closed.  Thank's for the support!
